fairmq/shmem/Socket.h

```diff
@@ -125,23 +135,83 @@ class Socket final : public fair::mq::Socket
         return zmq::Connect(fSocket, address, fId);
     }
 
-    int64_t Send(MessagePtr& msg, int timeout = -1) override
+  private:
+    auto MakeMetaMsg(shmem::Message const& shmMsg) noexcept -> zmq_msg_t
     {
+        std::size_t const size = std::max(fMetadataMsgSize, sizeof(MetaHeader));
+        zmq_msg_t metaMsg;
+        auto rc = 0;
+
+        if (fMetadataMsgSize > size) {
+            // copy the MetaHeader into a larger buffer than needed
+            auto paddedMetaMsg = tools::MakePaddedUnique<MetaHeader>(
+                size, alignof(std::max_align_t), std::pmr::get_default_resource(), shmMsg.fMeta);
+
+            using Deleter = typename decltype(paddedMetaMsg)::deleter_type;
+            auto del = std::make_unique<Deleter>(paddedMetaMsg.get_deleter());
```

Comment:
This extra allocation I would like to avoid still, was thinking of copying the deleter into the free buffer space after the `MetaHeader`. Just I am not sure, if this would be UB, if the deleter would deallocate itself "as last action" (and also what about calling the dtor?!).